include/envoy/http/codec.h

```diff
@@ -189,6 +210,9 @@ struct Http2Settings {
   // our default connection-level window also equals to our stream-level
   static const uint32_t DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE = 256 * 1024 * 1024;
   static const uint32_t MAX_INITIAL_CONNECTION_WINDOW_SIZE = (1U << 31) - 1;
+
+  // By default, do not enforce stream buffer limits beyond the connection level limits.
+  static const uint32_t DEFAULT_PER_STREAM_BUFFER_LIMIT = 0;
```

Comment:
I just looked at the code more, and correct me if I am wrong the default OOB behavior will be:

- 1MiB connection buffer limit, when breached, fires callbacks and pauses all streams.
- No default codec level buffer limit. If stream/connection windows back up, nothing will be paused and we can use unlimited memory. (In case TCP is open, but other side is not ACKing window updates).

IMO we should set buffer limit by default, and perhaps merge stream buffer/window settings into single setting?